datafusion/common/tests/macro_hygiene.rs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Could we possibly make this a module of an existing test (rather than a an entirely new test, which makes a new binary which at some point caused the CI runner to run out of disk space)?

For example, somewhere in https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/core_integration.rs maybe 🤔

Reply:
done